cover-icon-element.js

```diff
@@ -78,18 +81,20 @@ class CoverIconElement extends HTMLElement {
   }
 }
 
-CoverIconElement.dataImage4 = (color) => `data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='${encodeURIComponent(color)}' preserveAspectRatio='xMidYMid meet' focusable='false' viewBox='0 0 24 24'%3E%3Cg%3E%3Cpath d='M3 4H21V8H19V20H17V8H7V20H5V8H3V4'%3E%3C/path%3E%3Cpath d='M3 9H16V11H8V9' opacity='1'%3E%3C/path%3E%3Cpath d='M3 12H16V14H8V12' opacity='1'%3E%3C/path%3E%3Cpath d='M3 15H16V17H8V15' opacity='1'%3E%3C/path%3E%3Cpath d='M3 18H16V20H8V18' opacity='1'%3E%3C/path%3E%3C/g%3E%3C/svg%3E`;
+CoverIconElement.dataImage4 = (color, opacity) => `data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='${encodeURIComponent(color)}' preserveAspectRatio='xMidYMid meet' focusable='false' viewBox='0 0 24 24'%3E%3Cg%3E%3Cpath d='M3 4H21V8H19V20H17V8H7V20H5V8H3V4'%3E%3C/path%3E%3Cpath d='M3 9H16V11H8V9' opacity='1'%3E%3C/path%3E%3Cpath d='M3 12H16V14H8V12' opacity='1'%3E%3C/path%3E%3Cpath d='M3 15H16V17H8V15' opacity='1'%3E%3C/path%3E%3Cpath d='M3 18H16V20H8V18' opacity='1'%3E%3C/path%3E%3C/g%3E%3C/svg%3E`;
```

Comment:
I don't think you need to `encodeURIComponent`  the opacities since they can't contain invalid chars (unlike the `fill` property when a color is specified via a hex code - `#fefefe`).

Reply:
I did it because 'opacity' is also a keyword for svg images and wanted to be sure. But I'm not very keen on JS so maybe it's not necessary and can be removed.
What I wanted to add but forgot to is some sort of check on the opacity, to be sure it's a valid float between 0-1, but I think it just reverts to full black if it finds a "strange value"

Reply:
I normally use https://yoksel.github.io/url-encoder/ to encode the svg. Although I'm not sure it's necessary it is, let's keep it using it for now.

Here's a possibility:
```
CoverIconElement.dataImageError = (color) => `data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='${encodeURIComponent(color)}' preserveAspectRatio='xMidYMid meet' focusable='false' viewBox='0 0 24 24'%3E%3Cg%3E%3Cpath d='M3 4H21V8H19V20H17V8H7V20H5V8H3V4'%3E%3C/path%3E%3Cpath d='M10 9 H14 V15 H 10 Z' opacity='0.7'%3E%3C/path%3E%3Cpath d='M10 16 H14 V18.5 H 10 Z' opacity='0.7'%3E%3C/path%3E%3C/g%3E%3C/svg%3E`;
```
![Screenshot 2021-11-16 at 09 58 18](https://user-images.githubusercontent.com/1107697/141963782-6d64d227-2173-4c51-a9ff-49b9b3922401.png)

Reply:
There are pros and cons: the one I used is a "standard" mdi icon, this is more in line with the rest.
To my taste, I'd make the exclamation mark thinner as in the mdi icon...

Reply:
Thinner:
```
CoverIconElement.dataImageError = (color) => `data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='${encodeURIComponent(color)}' preserveAspectRatio='xMidYMid meet' focusable='false' viewBox='0 0 24 24'%3E%3Cg%3E%3Cpath d='M3 4H21V8H19V20H17V8H7V20H5V8H3V4'%3E%3C/path%3E%3Cpath d='M11 9.5 H13 V15 H 11 Z' opacity='0.9'%3E%3C/path%3E%3Cpath d='M11 16 H13 V18 H 11 Z' opacity='0.9'%3E%3C/path%3E%3C/g%3E%3C/svg%3E`;
```

![Screenshot 2021-11-16 at 11 32 49](https://user-images.githubusercontent.com/1107697/141978905-4d18d773-db14-4d86-af1c-2b075845198a.png)